### PR TITLE
Practice app/currency converter

### LINKED ISSUE #190: [Practice App]: Frontend for Currency Converting
Users can see the conversion rates between [USD, EUR, TRY, AZN] in a table format.

They can also calculate the value of some amount of currency in the other currency format.

Simple HTML with NextJs is used for the project.

Deadline: 19.05.2022 23.59
Commit: [commit](https://github.com/bounswe/bounswe2022group3/commit/9fb9edcfc3955753a29bb71021a1e307479f2fa6)

### LINKED ISSUE #191: [Practice App]: Backend for Currency Converting
Backend provides both a post and get request.
They can also calculate the value of some amount of currency in the other currency format.

ExpressJs is used. 
POST: `/currency/allConversions` --> Returns all exchange rates for [USD, EUR, TRY, AZN].
GET: `/currency/getConversion` --> Returns the value of a currency in the form of other in any amount.

Deadline: 19.05.2022 23.59
Commit: [commit](https://github.com/bounswe/bounswe2022group3/commit/9fb9edcfc3955753a29bb71021a1e307479f2fa6)

### LINKED ISSUE #192: [Practice App]: Unit Tests for Currency Converting
We should create unit tests to make sure that our endpoints work as expected.
For that, I will be using JEST and supertest, as they will help us easily mock the external api and make the testing descriptions a bit more intuitive.

Deadline: 19.05.2022 23.59
Commit: [commit](https://github.com/bounswe/bounswe2022group3/commit/b6e57ac500945c055bb7f064ada20985b40fe664)

### LINKED ISSUE #194: [Practice App]: Swagger API Documentation for Currency Converting
We should provide some documentation about our endpoints.
I will be documenting the endpoints relevant to currency converting feature:
GET `/currency/getConversion`
POST `/currency/allConversions`

Deadline: 19.05.2022 23.59

### PR DESCRIPTION
**Pull request by yilmaz-burak**
I have implemented a simple currency converter. It converts currencies between [USD, EUR, TRY, AZN] currencies.

Relevant backend, front-end and unit tests are implemented. Also added necessary env variables to .env.example file for the convenience.

Thanks!

resolves #194
resolves #193 
resolves #192 
resolves #191 
resolves #190

### CODE REVIEW

File: practice-app/server/src/app.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ app.use(
   })
 );
 
+app.use(cors());
```
Comment (bilalcim) [inline practice-app/server/src/app.js:16]: Where do you use this?

**Review by bilalcim — COMMENTED** [review]
You do not have a model, but this is because you do not use the database, and that is simply because the nature of things is volatility itself, right?

Also, about using the .env file for your API URL: Is there a specific reason why you don't include that in the code? I am trying to learn.

Your tests are quite thorough. One small thing about validation though: Is your validate.js file empty because you have covered those in testing? That does not sound quite right to me, so I am guessing there is another mechanism.

There are one or two things that I would just _prefer_ to do in another way, but all in all, I learned a lot from reviewing this pull request already!

**Comment by yilmaz-burak** [comment]
You have very valid points. Regarding especially the second and thirds comments of yours, I have made some additions. Thanks for the review! Please let me know if anything else needs a retouch!

**Review by bilalcim — APPROVED** [review]
Looks good to me!

**Pull request merged**